### ISSUE TITLE
Weird Behavior of PadTo and poor documentation

### ISSUE THREAD

**Issue by Optimox** [kornia/kornia#2121]
### Describe the bug

Hi,

First thank you for this great library, I'm currently trying to switch all my pipeline to kornia instead of albumentation.

I was actually looking for a PadIfNeeded function in Kornia.
I finally found that there exists a `PadTo` function (not available in the documentation website by the way).

However, the behavior of the function is unclear when you pass a larger image "inside a smaller `PadTo`" (see bellow).

There is also a small typo in the docstring saying : `p: probability of the image being flipped.`

### Reproduction steps

```bash
import kornia.augmentation as K
import torch
a = torch.meshgrid(torch.arange(10), torch.arange(10))[0].float()

transform = K.PadTo((5, 5), pad_value=0)

transform(a)
```

### Expected behavior

It seems that the transformation returns the upper left corner of the image, without any padding happening of course.

I guess I would expect two possibilities:
- raise an error: since the augmentation does not perform anything related to what it's supposed to do -> Error: PadSize should be larger than the input size.
- return the original image: this would be equivalent to a pad if needed function.

### Environment

```shell
kornia='0.6.9'
torch='1.12.1+cu113'
```

**Comment by johnnv1** [comment]
Thanks for the report, I will update the docs

my interpretation is that they have different behaviors, I think the most appropriate here would be to add as an extra option or even as a new augmentation. As a side note, this function "just" uses `torch.nn.functional.pad`.

cc: @shijianjian

**Comment by Optimox** [comment]
Yes I agree that `PadTo` does not have to be similar to `PadIfNeeded` however returning the upper left corner without any warning of any form does not seem right to me.

The problem comes from this line : https://github.com/kornia/kornia/blob/3bf798ee6d5e1cfdb1dcca2fa7959e6acd3007b1/kornia/augmentation/_2d/geometric/pad.py#L68 which can create a negative `height_pad`. I actually find it strange that `torch.nn.functional.pad` accepts that.

If it can help here is a sketch of what I would expect from `PadIfNeeded` (please note that there is no correct docstring, the inverse_tranform is not implemented and I don't know what is wrong with apply transform method). Also I slightly changed the behavior of the padding to get a central image as it seems more natural to me to center the original image:

```
from typing import Any, Dict, Optional, Tuple, Union, cast
from kornia.augmentation._2d.geometric.base import GeometricAugmentationBase2D
from torch import Tensor
import torch
class PadIfNeeded(GeometricAugmentationBase2D):
    r"""Pad the given sample to a specific size only if needed.
    """

    def __init__(
        self,
        size: Tuple[int, int],
        pad_mode: str = "constant",
        pad_value: Union[int, float] = 0,
        keepdim: bool = False,
        return_transform: Optional[bool] = None,
    ) -> None:
        super().__init__(p=1.0, return_transform=return_transform, same_on_batch=True, p_batch=1.0, keepdim=keepdim)
        self.flags = dict(size=size, pad_mode=pad_mode, pad_value=pad_value)

    # REMARK: I don't know what those TODOs are but it's worth doing them:)
    # TODO: It is incorrect to return identity
    # TODO: Having a resampled version with ``warp_affine``
    def compute_transformation(self, image: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any]) -> Tensor:
        return self.identity_matrix(image)

    def apply_transform(
        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
    ) -> Tensor:
        _, _, height, width = input.shape

        height_pad: int = max(flags["size"][0] - height, 0)
        width_pad: int = max(flags["size"][1] - width, 0)
        
        # center the padding
        left_width: int = width_pad // 2
        right_width: int = width_pad - left_width
            
        top_height: int = height_pad // 2
        bottom_height: int = height_pad - top_height
            
        return torch.nn.functional.pad(
            input, [left_width, right_width, top_height, bottom_height], mode=flags["pad_mode"], value=flags["pad_value"]
        )

    def inverse_transform(
        self,
        input: Tensor,
        flags: Dict[str, Any],
        transform: Optional[Tensor] = None,
        size: Optional[Tuple[int, int]] = None,
    ) -> Tensor:
        size = cast(Tuple[int, int], size)
        raise NotImplementedError
```

**Comment by johnnv1** [comment]
yep yep, i gotcha. As I said, to me it seems to be usual to guarantee the desired size, otherwise it can end up breaking aug pipelines or become verbose. In my view, I think it would be a case of just adding it as a note in the documentation. And if Jian thinks it's ok we could update it so that the padding is the same for all sides (for me is also the most natural scenario).

**Comment by shijianjian** [comment]
I think we do have padifneeded function in RandomCrop. 

I cannot fully recall the issue with PadTo, since we need to handle more conditions like the inverse, the augmentation chain, different data keys, and a chain of inverse, etc. As a result, we tended to only pad the right bottom sides.

Any PRs are welcomed as long as it handles all conditions nicely. 

I agree that we should improve more regarding the documentation and tutorials. Like @johnnv1 did already. :)

**Comment by shijianjian** [comment]
I think the `PadTo` shall be updated if anyone interested.

Basically, it shall be re-written with `NonRigidAugmentation` base class, then to define all the transformations (e.g. different data keys, inverses) by hand.
